### PR TITLE
fix: Sort operator versions by semver version

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Before
```bash
stackablectl op desc airflow                          

 OPERATOR         airflow                                                                                                                           
 STABLE VERSIONS  0.1.0, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 23.1.0, 23.11.0, 23.4.0, 23.4.1, 23.7.0, 24.11.0, 24.3.0, 24.7.0                        
 TEST VERSIONS    0.0.0-pr517, 0.0.0-pr518, 0.0.0-pr519, 0.0.0-pr520, 0.0.0-pr521, 0.0.0-pr524, 0.0.0-pr525, 0.0.0-pr526, 0.0.0-pr527, 0.0.0-pr529, 
                  0.0.0-pr530, 0.0.0-pr531, 0.0.0-pr532, 0.0.0-pr533, 0.0.0-pr534, 0.0.0-pr537, 0.0.0-pr538, 0.0.0-pr539, 0.0.0-pr540               
 DEV VERSIONS     0.0.0-dev
```

After
```bash
cargo r -p stackablectl -- op desc airflow

 OPERATOR         airflow                                                                                                                           
 STABLE VERSIONS  0.1.0, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 23.1.0, 23.4.0, 23.4.1, 23.7.0, 23.11.0, 24.3.0, 24.7.0, 24.11.0                        
 TEST VERSIONS    0.0.0-pr517, 0.0.0-pr518, 0.0.0-pr519, 0.0.0-pr520, 0.0.0-pr521, 0.0.0-pr524, 0.0.0-pr525, 0.0.0-pr526, 0.0.0-pr527, 0.0.0-pr529, 
                  0.0.0-pr530, 0.0.0-pr531, 0.0.0-pr532, 0.0.0-pr533, 0.0.0-pr534, 0.0.0-pr537, 0.0.0-pr538, 0.0.0-pr539, 0.0.0-pr540               
 DEV VERSIONS     0.0.0-dev
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Review by Techassi — DISMISSED** [review]

**Comment by Techassi** [comment]
The pre-commit error can be fixed by installing the Node dependencies. @NickLarsenNZ do we want to do this now, or force merge to get this fix into the 24.11.0 release of stackablectl?

**Comment by NickLarsenNZ** [comment]
> The pre-commit error can be fixed by installing the Node dependencies. @NickLarsenNZ do we want to do this now, or force merge to get this fix into the 24.11.0 release of stackablectl?

What are the tradeoffs of doing it now or leaving it until after the release?
(Sorry, brain is in a different place right now).

**Comment by Techassi** [comment]
I see two options:

- Don't merge this for the 24.11.0 release and instead merge it afterwards to create a 24.11.1 release
- Merge it (and release it in 24.11.0) either by
  - ignoring the check and force-merging it. Fix the error afterwards in a separate PR.
  - fixing the issue in this PR.

**Comment by NickLarsenNZ** [comment]
> I see two options:
> 
>     * Don't merge this for the 24.11.0 release and instead merge it afterwards to create a 24.11.1 release
> 
>     * Merge it (and release it in 24.11.0) either by
>       
>       * ignoring the check and force-merging it. Fix the error afterwards in a separate PR.
>       * fixing the issue in this PR

I would tend to go with leaving it out to save all the extra work, as I'm unsure how critical this is right now. Happy to let it go in if someone says they really want it in the release though.

**Comment by Techassi** [comment]
Alright, then we leave it open for now and merge it for a 24.11.1 release of stackablectl.

**Comment by sbernauer** [comment]
pre-commit action fixed and green :heavy_check_mark:

File: rust/stackablectl/src/cmds/operator.rs [hunk 1]
```diff
@@ -130,6 +130,12 @@ pub enum CmdError {
     #[snafu(display("invalid repository name"))]
     InvalidRepoName { source: InvalidRepoNameError },
 
+    #[snafu(display("invalid helm chart version {version:?} (need to be semver)"))]
```
Comment (Techassi) [inline rust/stackablectl/src/cmds/operator.rs]: suggestion: Slightly reword error message.

 ```suggestion
    #[snafu(display("invalid semantic helm chart version {version:?}"))]
```

File: rust/stackablectl/src/cmds/operator.rs [hunk 2]
```diff
@@ -524,16 +530,15 @@ where
         Some(entries) => {
             let mut versions = entries
                 .iter()
-                .map(|e| Version::parse(&e.version))
-                .map_while(|r| match r {
-                    Ok(v) => Some(v),
-                    Err(_) => None,
+                .map(|entry| {
+                    Version::parse(&entry.version).context(InvalidHelmChartVersionSnafu {
```
Comment (Techassi) [inline rust/stackablectl/src/cmds/operator.rs]: note: We could use `with_context` to only clone the string when an error occurs.

```suggestion
                    Version::parse(&entry.version).with_context(|_| InvalidHelmChartVersionSnafu {
```

File: README.md [hunk 3]
```diff
@@ -77,4 +77,3 @@ hooks are:
 [pre-commit]: https://pre-commit.com/
 [web-readme]: ./web/README.md
 [lib-readme]: ./rust/stackable-cockpit/README.md
-[xtasks]: ./xtask/src/main.rs
```
Comment (Techassi) [inline README.md:80]: question: I assume this was unused and therefore removed?

File: web/build.rs [hunk 4]
```diff
@@ -16,18 +16,21 @@ fn main() {
...
         .arg(&vite_out_dir)
         .arg("--base")
-        .arg("/ui/")
-        .status()
-        .unwrap();
-    if !vite_status.success() {
-        panic!("web-ui build failed: {vite_status}");
-    }
+        .arg("/ui/");
+    let vite_status = vite_command.status();
```
Comment (Techassi) [inline web/build.rs]: suggestion: Add a newline here to give a little more breathing room.

```suggestion
        .arg("/ui/");

    let vite_status = vite_command.status();
```

**Review by Techassi — CHANGES_REQUESTED** [review]
LGTM, just some minor comments.

File: README.md (hunk 3 above)
Comment (sbernauer) [reply README.md:80]: Exactly, pre-commit complained about it being unused

**Pull request merged**